### PR TITLE
Add example fields function #219

### LINKED ISSUE #219: Missing a better example for the table resources schema
The schema.json only provides a single example. We have a nice example available in the previous versions. There is only the nullable missing. We could add some extra code to extend the example and add it to the example.json. We are missing a better example for the table resources schema. 

_Originally posted by @jh-RLI in https://github.com/OpenEnergyPlatform/oemetadata/issues/216#issuecomment-2470948567_

### PR DESCRIPTION
**Pull request by Ludee**
## Summary of the discussion

The schema.json only provides a single example. We have a nice example available in the previous versions. There is only the nullable missing. We could add some extra code to extend the example and add it to the example.json. We are missing a better example for the table resources schema.

Originally posted by @jh-RLI in https://github.com/OpenEnergyPlatform/oemetadata/issues/216#issuecomment-2470948567

### Added
- Add example fields function [(#220)](https://github.com/OpenEnergyPlatform/oemetadata/pull/220)

### Automation
Closes #219

### PR-Assignee
- [x] 🐙 Follow the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/oemetadata/blob/develop/CONTRIBUTING.md)
- [x] 📝 Update the [CHANGELOG.md](https://github.com/OpenEnergyPlatform/oemetadata/blob/develop/CHANGELOG.md)
- [x] 📙 Update the documentation
- [x] 🐙 Assign a reviewer to the PR

### Reviewer
- [x] 🐙 Follow the [Reviewer Guidelines](https://github.com/OpenEnergyPlatform/oemetadata/blob/develop/CONTRIBUTING.md#40-let-someone-else-review-your-pr)
- [x] 🐙 Provided feedback and show sufficient appreciation for the work done

### CODE REVIEW

File: metadata/v20/v20/build_source/scripts/example_fields.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    {
+        "name": "geom",
+        "description": "Geometry",
+        "type": "geometry(Point, 4326)",
+        "nullable": true,
+        "unit": null,
+        "isAbout": [
+            {
+                "name": "spatial region",
+                "path": "https://openenergy-platform.org/ontology/oeo/BFO_0000006"
```
Comment (jh-RLI) [inline metadata/v20/v20/build_source/scripts/example_fields.json]: the "path" in these fields should be "@id". This reflects the latest updates i think.

File: metadata/v20/v20/build_source/scripts/generate_example_from_schema.py [hunk 2]
```diff
@@ -215,11 +215,42 @@ def test_oemetadata_schema_should_validate_oemetadata_example(example):
...
+    # Open and read the JSON file
+    with open(file_path, 'r') as file:
+        data = json.load(file)
+
+    # Perform the key replacement
+    if find_and_replace_key(data, target_key, new_value):
+        # Save the updated JSON data back to the file
+        with open(file_path, 'w') as file:
+            json.dump(data, file, indent=4)
+        print(f"Updated '{target_key}' to '{new_value}' in {file_path}")
```
Comment (jh-RLI) [inline metadata/v20/v20/build_source/scripts/generate_example_from_schema.py]: minor thing: using logger.info() here is nicer :)

File: metadata/v20/v20/build_source/scripts/generate_example_from_schema.py [hunk 3]
```diff
@@ -215,11 +215,42 @@ def test_oemetadata_schema_should_validate_oemetadata_example(example):
...
+def replace_key_in_json(file_path, target_key, new_value):
+    # Open and read the JSON file
+    with open(file_path, 'r') as file:
+        data = json.load(file)
+
+    # Perform the key replacement
+    if find_and_replace_key(data, target_key, new_value):
+        # Save the updated JSON data back to the file
+        with open(file_path, 'w') as file:
+            json.dump(data, file, indent=4)
```
Comment (jh-RLI) [inline metadata/v20/v20/build_source/scripts/generate_example_from_schema.py]: here we need to ensure the UTF-8 encoding to avoide strange characters in the result:

``` python
with open(file_path, "w", encoding="utf-8") as file:
    json.dump(data, file, ensure_ascii=False, indent=4)
```

**Review by jh-RLI — CHANGES_REQUESTED** [review]
once the changes with the UTF-8 encoding is done please regenerate the example :)

File: metadata/v20/v20/build_source/scripts/example_fields.json (hunk 1 above)
Comment (Ludee) [reply metadata/v20/v20/build_source/scripts/example_fields.json]: fixed

File: metadata/v20/v20/build_source/scripts/generate_example_from_schema.py (hunk 2 above)
Comment (Ludee) [reply metadata/v20/v20/build_source/scripts/generate_example_from_schema.py]: fixed

File: metadata/v20/v20/build_source/scripts/generate_example_from_schema.py (hunk 3 above)
Comment (Ludee) [reply metadata/v20/v20/build_source/scripts/generate_example_from_schema.py]: fixed

**Comment by jh-RLI** [comment]
Hm for some reason my requested changes do not resolve the issue with the unexpected symbol in special strings.

**Comment by jh-RLI** [comment]
Okay i think you might have some issue on your machine, maybe you open this BOM in the json. When i use the script to regenerate the example.json the symbols are correct. Still, there appears to be a strange issue with the metadata import that is raised after the generation is complete.

```bash
#The generation is done and then this exception is raised:

ntology/oeo/OEO_00000350'}], 'valueReference': [{'value': None, 'name': None, '@id': None}]}, {'name': 'geom', 'description': 'Geometry', 'type': 'geometry(Point, 4326)', 'nullable': True, 'unit': None, 'isAbout': [{'name': 'spatial region', '@id': 'https://openenergy-platform.org/ontology/oeo/BFO_0000006'}], 'valueReference': [{'value': None, 'name': None, '@id': None}]}]' in metadata/v20/v20/example.json
Traceback (most recent call last):
  File "/home/jh/github/oemetadata/metadata/v20/v20/build_source/scripts/generate_example_from_schema.py", line 262, in <module>
    test_oemetadata_schema_should_validate_oemetadata_example(json_data)
  File "/home/jh/github/oemetadata/metadata/v20/v20/build_source/scripts/generate_example_from_schema.py", line 214, in test_oemetadata_schema_should_validate_oemetadata_example
    from metadata.v20.v20.schema import OEMETADATA_V20_SCHEMA
ModuleNotFoundError: No module named 'metadata.v20'
```

**Pull request merged**